### DISCUSSION TITLE
New devices - update with zigbee definition file

### DISCUSSION THREAD

**Discussion by andypmuc** [arendst/Tasmota#17222] [Q&A]
Hello,

I bought Sonoff Zigbee Bridge Pro flashed tasmota and thought I could attach nearly all Zigbee devices. Sorry, my first time. Then I bought Nous E6 Temperature & Humidity sensors, but I didn't get it running. In the beginning it worked, but after a few hours it stopped transmitting.

Then I found the https://zigbee.blakadder.com/zigbee2tasmota.html pages and saw, that the device is not compatible.

Now the question, I saw the zigbee definition files. Could I add the sensor, by uploading a new definition file? Like it is done here https://zigbee.blakadder.com/Moes_ZSS-KB-TH.html

Or is this a completely different story?

Thanks

**Comment by sfromis** [comment]
Assuming that you mean https://zigbee.blakadder.com/Nous_E6.html that page does not say that is is not compatible, only that it has not (yet) been confirmed to be working with Tasmota, that is very common due to the large number of available Zigbee devices, and many of them would most likely work out of the box, but maybe not this one..... Exactly what it would take to make it work depends on details (which is not my domain), but *might* include such a definition file.

**Comment by andypmuc** [comment]
OK, I have done some testing, it would seem the device is falling in sleep mode. Wenn I press the button on the device, it sends an update, but as far as I can tell no update, when temp or humi changes, which is wanted ...
Maybe someone with more experience can help ...

**Comment by s-hadinger** [comment]
Can you force a new pairing and copy the logs of the pairing here? It should give us additional information about the device.

**Comment by andypmuc** [comment]
Now it seems to work and I do not want to separate it now, maybe later. 
It is also the case, that the sensor only changes temperature values every 0.6 degrees or when you press the top button. Maybe you can change this by a setting or a zigbee command, I found there something and have to investigate. Maybe the config in the info.
Here is the result of the 

{"ZbInfo":{"0x7DC5":{"Device":"0x7DC5","IEEEAddr":"0xA4C138368BB0EE6D","ModelId":"TS0601","Manufacturer":"_TZE200_nnrfa68v","Endpoints":[1],"Config":["~01.1"],"Reachable":true,"LastSeen":945,"LastSeenEpoch":1669966685,"LinkQuality":105}}}

**Comment by andypmuc** [comment]
Here it is https://nous.technology/product/e6/eng.html
Now I need to find a way, to send the command to the sensor without nous hub.

... ZbSend and then this ...

https://tasmota.github.io/docs/Zigbee/#sending-device-commands

Has someone experience here?

**Comment by andypmuc** [comment]
11:40:15.936 CMD: ZbSend {"Device":"0x7DC5","Write":{"Config":["~01.1"]}}
11:40:15.953 MQT: SonOffZigbeeBridge/stat/RESULT = {"ZbSend":"Unknown attribute 'Config'"}

Does this come from the sensor or Tasmota?

**Reply by s-hadinger** [reply]
'Config' is only internal Tasmota information. What command do you want to send?

You can change attribute reporting if you want it to report for smaller change, but this can deplete the battery faster. Not all device actually accept changing those parameters.

It's not yet in the doc but described in the PR here https://github.com/arendst/Tasmota/pull/9050

**Comment by andypmuc** [comment]
I now tried to send it during pairing, but it does not work. Also tried console and MQTT command. Can somebody help?

You can change the interval according to manual of the sensor.

ZbSend {"Device":"0x8932","ReadConfig":{"Power":true}}
ZbSend {"Device":"0x885E","ReadConfig":{"Temperature":{"MinInterval":true}}}

Log says

10:46:42.420 CMD: ZbPermitJoin 1
10:46:42.429 MQT: SonOffZigbeeBridge/tele/RESULT = {"ZbState":{"Status":21,"Message":"Pairing mode enabled"}}
10:46:42.452 MQT: SonOffZigbeeBridge/stat/RESULT = {"ZbPermitJoin":"Done"}
10:46:54.607 MQT: SonOffZigbeeBridge/tele/RESULT = {"ZbState":{"Status":34,"IEEEAddr":"0xA4C138368BB0EE6D","ShortAddr":"0x885E","ParentNetwork":"0x0000"}}
10:46:55.107 MQT: SonOffZigbeeBridge/tele/RESULT = {"ZbState":{"Status":30,"IEEEAddr":"0xA4C138368BB0EE6D","ShortAddr":"0x885E","PowerSource":false,"ReceiveWhenIdle":false,"Security":false}}
10:46:55.204 MQT: SonOffZigbeeBridge/tele/RESULT = {"ZbState":{"Status":32,"ActiveEndpoints":["0x01"]}}
10:46:55.512 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","ModelId":"TS0601","Manufacturer":"_TZE200_nnrfa68v","Endpoint":1,"LinkQuality":7}}}
10:46:57.002 MQT: SonOffZigbeeBridge/tele/RESULT = {"ZbState":{"Status":33,"Device":"0x885E","Endpoint":"0x01","ProfileId":"0x0104","DeviceId":"0x0051","DeviceVersion":1,"InClusters":["0x0004","0x0005","0xEF00","0x0000"],"OutClusters":["0x0019","0x000A"]}}
10:46:59.629 ZIG: Zigbee Devices Data saved in File System (279 bytes)
10:47:01.935 ZIG: Battery auto-probe `ZbSend {"Device":"0x885E","Read":{"BatteryPercentage":true,"BatteryVoltage":true}}`
10:47:01.945 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","EF00?11":"0FE040","Endpoint":1,"LinkQuality":51}}}
10:47:02.140 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","AppVersion":66,"0000/FFE2":51,"0000/FFE4":0,"Endpoint":1,"LinkQuality":47}}}
10:47:02.431 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","0000/FFDF":"�rE+f�rE+","Endpoint":1,"LinkQuality":47}}}
10:47:02.920 ZIG: Auto-responder: ZbSend {"Device":"0x885E","Cluster":"0x000A","Endpoint":1,"Response":"000A/0007":725971622}
10:47:02.931 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","Cluster":10,"Read":[7],"ReadNames":{"LocalTime":true},"Endpoint":1,"LinkQuality":47}}}
10:47:05.674 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","EF00/0409":0,"Endpoint":1,"LinkQuality":47}}}
10:47:05.821 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","EF00/020A":0,"Endpoint":1,"LinkQuality":47}}}
10:47:05.971 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","EF00/020B":0,"Endpoint":1,"LinkQuality":51}}}
10:47:06.324 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","EF00/0213":6,"Endpoint":1,"LinkQuality":47}}}
10:47:06.473 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","EF00/0204":100,"Endpoint":1,"LinkQuality":47}}}
10:47:06.675 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","EF00/0201":207,"Endpoint":1,"LinkQuality":51}}}
10:47:06.823 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","EF00/0202":73,"Endpoint":1,"LinkQuality":47}}}
10:47:07.023 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","EF00/0409":0,"Endpoint":1,"LinkQuality":51}}}
10:47:09.469 MQT: SonOffZigbeeBridge/stat/RESULT = {"ZbSend":"Done"}
10:47:10.115 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","EF00?24":"7410","Endpoint":1,"LinkQuality":51}}}
10:47:15.116 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","EF00?24":"7510","Endpoint":1,"LinkQuality":51}}}
10:47:20.097 MQT: SonOffZigbeeBridge/tele/SENSOR = {"ZbReceived":{"0x885E":{"Device":"0x885E","EF00?24":"7610","Endpoint":1,"LinkQuality":51}}}
10:47:32.387 MQT: SonOffZigbeeBridge/stat/RESULT = {"ZbSend":"Done"}
10:47:42.471 MQT: SonOffZigbeeBridge/tele/RESULT = {"ZbState":{"Status":20,"Message":"Pairing mode disabled"}}
10:47:43.022 MQT: SonOffZigbeeBridge/stat/RESULT = {"ZbSend":"Done"}

**Reply by s-hadinger** [reply]
Are you trying random commands? I have no idea what you are trying to do.

**Reply by andypmuc** [reply]
No, I have paired the device, to ensure it is online and then
send over mqtt
ZbSend {"Device":"0x8932","ReadConfig":{"Power":true}}
received
10:47:32.387 MQT: SonOffZigbeeBridge/stat/RESULT = {"ZbSend":"Done"}
send
ZbSend {"Device":"0x885E","ReadConfig":{"Temperature":{"MinInterval":true}}}
received
10:47:43.022 MQT: SonOffZigbeeBridge/stat/RESULT = {"ZbSend":"Done"}

But nothing more, that's the problem...

**Reply by sfromis** [reply]
Try with `Weblog 3` to see if you get details then....

**Reply by andypmuc** [reply]
Now I get this

22:09:02.204 CMD: ZbSend {"Device":"0x8932","ReadConfig":{"Power":true}}
22:09:02.206 SRC: WebConsole from 192.168.1.27
22:09:02.209 CMD: Grp 0, Cmd 'ZBSEND', Idx 1, Len 47, Pld -99, Data '{"Device":"0x8932","ReadConfig":{"Power":true}}'
22:09:02.212 ZIG: guessing endpoint 0
22:09:02.218 MQT: SonOffZigbeeBridge/stat/RESULT = {"ZbSend":"Missing endpoint"}

**Reply by andypmuc** [reply]
and this, I have renamed it

22:13:22.547 CMD: zbinfo 0x8932
22:13:22.555 MQT: SonOffZigbeeBridge/stat/RESULT = {"ZbInfo":"Unknown device"}
22:13:52.218 CMD: zbinfo THS
22:13:52.226 MQT: SonOffZigbeeBridge/stat/RESULT = {"ZbInfo":"Unknown device"}

but it is on the map

**Reply by Woody14619** [reply]
What you probably want is to configure the interval and reporting differences.  For most things this would be done automatically, but for whatever reason, it's not working on your device.

Make sure the button is awake, and attempt this:
```code
ZbSend {"Devince":"0x8932","Config":{"Temperature":{"MinInterval":30,"MaxInterval":3600,"ReportableChange":0.5}}}
```

That should cause the device to report temperatures only once an hour, or if the temperature changes more than half a degree (Celsius).  At most it will report twice a minute, assuming the temperature is changing that quickly.  You can set similar restrictions for battery percentage and voltage.

**Reply by Woody14619** [reply]
Also, if you get this reliably working, using the config line above or otherwise, you should update the page to indicate it's compatible with Z2Tasmota.  If you know it's working, but aren't sure how to update it, I could update if for you.  If there are specific config lines to get it working, I can add those as well if you include them here. (Or, if you prefer, you can update the page directly yourself.  Just figured I'd offer, since there's a bit of a learning curve to updating the pages, as it's all done via `git`.
